pimd/pim_tib.c

```diff
@@ -35,7 +35,11 @@ tib_sg_oil_setup(struct pim_instance *pim, pim_sgaddr sg, struct interface *oif)
 	up = pim_upstream_find(pim, &sg);
 	if (up) {
 		memcpy(&nexthop, &up->rpf.source_nexthop, sizeof(struct pim_nexthop));
-		pim_nht_lookup_ecmp(pim, &nexthop, vif_source, &grp, false);
+		if (!pim_nht_lookup_ecmp(pim, &nexthop, vif_source, &grp, false))
+			if (PIM_DEBUG_PIM_NHT_RP)
+				zlog_debug("%s: Nexthop Lookup failed (%pPA, %pPA)", __func__,
+					   &sg.src, &sg.grp);
```

Comment:
This should be vif_source instead of sg.src.

Reply:
To further clarify, the vif_source will be set to whatever the address is that you are looking for the nexthop. Which could be the sg.src, or if that is not set, then it could be the RP/Upstream for the sg.grp.
So including both the vif_source and the sg pair in the debug log may be useful.